src/components/common/Sidebar/Sidebar.js

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from 'react';
+import 'antd/dist/antd.css';
+import '../styles/Sidebar.css';
+import { Layout, Menu, Switch as Toggle } from 'antd';
+import {
+  QuestionCircleOutlined,
+  DollarOutlined,
+  BulbOutlined,
+  CalendarOutlined,
+  ContainerOutlined,
+  UserOutlined,
+} from '@ant-design/icons';
+
+import {
+  MenteeComponents,
+  MentorComponents,
+  AdminComponents,
+} from './SidebarComponents';
+const { Content, Sider } = Layout;
+const { SubMenu } = Menu;
+
+const Sidebar = props => {
+  const { authService } = props;
+
+  const [collapsed, setCollapsed] = useState(false);
+  const [render, updateRender] = useState(1);
+
+  const onCollapse = collapsed => {
+    setCollapsed(collapsed);
+  };
+
+  const handleLogout = checked => {
+    authService.logout();
+    localStorage.removeItem('role_id');
+  };
+
+  const handleMenuClick = menu => {
+    updateRender(menu.key);
+  };
+
+  const role = localStorage.role_id;
+
+  const isUserMentee = () => {
+    if (role === '4') {
+      return true;
+    }
+    return false;
+  };
+
+  const isUserMentor = () => {
+    if (role === '3') {
+      return true;
+    }
+    return false;
+  };
+
+  const isUserAdmin = () => {
+    if (role === '2') {
+      return true;
+    }
+    return false;
+  };
+
+  return (
+    <Layout style={{ minHeight: '100vh' }}>
```

Comment:
I'm loving how you set up conditional component rendering.

Reply:
Seconding this - this was a really efficient way to implement dynamic rendering!